src/Symfony/Component/Messenger/Event/AbstractHandlerEvent.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Event;
+
+use Symfony\Component\Messenger\Envelope;
+
+abstract class AbstractHandlerEvent
+{
+    private Envelope $envelope;
+    private string $handlerName;
+
+    public function __construct(Envelope $envelope, string $handlerName)
+    {
+        $this->envelope = $envelope;
+        $this->handlerName = $handlerName;
+    }
```

Comment:
Please use constructor property promotion

Reply:
Yes! A couple questions:

- as we target PHP 8.1, do we want to keep `private` + getters, or can we use `public readonly` instead? (and if targeting Symfony 7 and PHP 8.2, straight `readonly class` with public promoted properties)
- does Symfony allow a mix of promoted and non-promoted, like this:
  ```php
  final class HandlerFailureEvent extends AbstractHandlerEvent
  {
      public function __construct(
          Envelope $envelope,
          string $handlerName,
          public \Throwable $exception,
      ) {
          parent::__construct($envelope, $handlerName);
      }
  }
  ```

Reply:
I've settled on `public readonly` as I could find other occurrences in the `7.0` branch. I also used a mix of promoted and non-promoted in `HandlerFailureEvent`, as in the code sample above.